batcher/message_batcher.go

```diff
@@ -0,0 +1,145 @@
+package batcher
+
+import (
+	"fmt"
+	"math/big"
+	"time"
+)
+
+type Sync interface {
+	SendBatch(batch [][]byte, largestSeq *big.Int, largetsSubSeq int)
+}
+
+type Batcher interface {
+	// Interval at which accumulated messages should be bulk put to
+	// firehose (default 1 second).
+	FlushInterval(dur time.Duration)
+	// Number of messages that triggers a push to firehose
+	// default to 10
+	FlushCount(count int)
+	// Size of batch that triggers a push to firehose
+	// default to 1mb (1024 * 1024)
+	FlushSize(size int)
+
+	//AddMesage to the batch
+	AddMessage(msg []byte, sequenceNumber string, subSequenceNumber int) error
+
+	//Flush all messages from the batch
+	Flush()
+}
+
+type batcher struct {
+	flushInterval time.Duration
+	flushCount    int
+	flushSize     int
+
+	largestSeq    *big.Int
+	largestSubSeq int
```

Comment:
not clear what these variables store... at what stage of processing do we mark these values?